cypress/e2e/downloads.cy.ts

```diff
@@ -1,354 +1,309 @@
-      return loadPlayer()
-        .then(() => {
-          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-          // @ts-ignore
-          cy.stub(player, 'isImage').returns(true);
-          setMedia();
-          return player?.ready();
-        })
-        .then(() => {
-          expect(requestCount).to.equal(0);
-        });
-    });
-  });
-
-  describe('download xhr call', () => {
-    let requestUrl = '';
-
-    beforeEach(() => {
-      cy.intercept('**/playManifest/**', req => {
-        ++requestCount;
-        requestUrl = req.url;
-        req.reply({
-          fixture: 'video.mp4'
-        });
-      }).as('playManifest');
-    });
-
-    afterEach(() => {
-      requestUrl = '';
-    });
-
-    describe('cdn url', () => {
-      it('should use cdn url from config if cdn url is set', async () => {
-        await loadPlayerAndSetMedia();
-        expect(requestUrl.indexOf(PLAYER_CONFIG.provider.env.cdnUrl)).to.not.equal(-1);
-      });
-      it('should not make download xhr call if cdn url is not set', async () => {
-        await loadPlayerAndSetMedia({provider: {env: {cdnUrl: ''}}});
-        expect(requestCount).to.equal(0);
+    it('should not exist for DRM media', () => {
+      mockKalturaBe();
+      loadPlayer().then(player => {
+        cy.stub(player, 'getVideoElement').returns({mediaKeys: {}});
+        setMedia(player);
+        cy.get('.playkit-pre-playback-play-button').should('exist').click({force: true});
+        cy.get('[data-testid="download-overlay-button"]').should('not.exist');
       });
     });
-    describe('partner id', () => {
-      it('should contain partner id from config if partner id is set', async () => {
-        await loadPlayerAndSetMedia();
-        expect(requestUrl.indexOf(`p/${PLAYER_CONFIG.provider.partnerId}`)).to.not.equal(-1);
-      });
-      it('should not make download call if partner id is not set', async () => {
-        await loadPlayerAndSetMedia({provider: {env: {partnerId: ''}}});
-        expect(requestCount).to.equal(0);
+    it('should be visible for non-DRM vod', () => {
+      mockKalturaBe();
+      loadPlayerAndSetMedia().then(() => {
+        cy.get('.playkit-pre-playback-play-button').should('exist').click({force: true});
+        cy.get('[data-testid="download-overlay-button"]').should('exist');
       });
     });
-    describe('entry id', () => {
-      it('should use current entry id', async () => {
-        await loadPlayerAndSetMedia();
-        expect(requestUrl.indexOf('entryId/1234')).to.not.equal(-1);
+    it('should show overlay on click', () => {
+      mockKalturaBe();
+      loadPlayerAndSetMedia().then(() => {
+        cy.get('.playkit-pre-playback-play-button').should('exist').click({force: true});
+        cy.get('[data-testid="download-overlay"]').should('not.exist');
+        cy.get('[data-testid="download-overlay-button"]').click({force: true});
+        cy.get('[data-testid="download-overlay"]').should('exist');
```

Comment:
i think it's better to test for "visible" because i think that "exist" can return true for elements that are in the dom but aren't visible
react won't add elements to the dom in some cases but i still think that visible is a more thorough test

Reply:
i think i tried visible but it failed for it because some parent element with overflow hidden or something..